teams/teams-ps/teams/Set-TeamsEnvironmentConfig.md

```diff
@@ -0,0 +1,141 @@
+---
+external help file: Microsoft.TeamsCmdlets.PowerShell.Connect.dll-Help.xml
+Module Name: MicrosoftTeams
+online version:
+schema: 2.0.0
+author: VikneshMSFT
+ms.author: vimohan
+ms.reviewer: pbafna
+manager: vinelap
+---
+
+# Set-TeamsEnvironmentConfig
+
+## SYNOPSIS
+
+Sets environment-specific configurations on the local machine and is used to connect to the right environment when running Connect-MicrosoftTeams.
+
+## SYNTAX
+
+```
+Set-TeamsEnvironmentConfig [-EndpointUris <Hashtable>] [-TeamsEnvironmentName <String>] [-WhatIf] [-Confirm]
+ [<CommonParameters>]
+```
+
+## DESCRIPTION
+This cmdlet sets environment-specific configurations like endpoint URIs(such as Azure AD and Microsoft Graph) and Teams environment (such as GCCH and DOD) on the local machine.
+  
+When running Connect-MicrosoftTeams, environment-specific information set in this cmdlet will be considered unless overridden by Connect-MicrosoftTeams parameters.
+
+Parameters passed to Connect-MicrosoftTeams will take precedence over the information set by this cmdlet. 
+
+Clear-TeamsEnvironmentConfig should not be used in Commercial, GCC, GCC High, or DoD environments.
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> Set-TeamsEnvironmentConfig -TeamsEnvironmentName TeamsChina
+```
+
+Sets the environment as Gallatin China on a local machine and when Connect-MicrosoftTeams is run, authentication will happen in the Gallatin China cloud and Microsoft Teams module will connect to the Gallatin environment.
+
+### Example 2
+```powershell
+$endPointUriDict = @{ActiveDirectory = 'https://login.microsoftonline.us/';MsGraphEndpointResourceId = 'https://graph.microsoft.us'}
+Set-TeamsEnvironmentConfig -TeamsEnvironmentName $endPointUriDict
+```
+Sets endpoint URIs required for special clouds.
+
+### Example 3
+```powershell
+Set-TeamsEnvironmentConfig -TeamsEnvironmentName TeamsChina
+
+$cred=get-credential
```

Comment:
Move-CsUser is an SFB on-prem cmdlet and it works with modern authentication. It internall calls Connect-MicrosoftTeams in MicrosoftTeams module. This change is to make a generic way to call move-csuser for Gallatin, Airgap and future special clouds, instead of tightly coupling, SFB on-prem CU release and MicrosoftTeams module. We wanted a way to decouple these components, so we need not release new SFB on-prem CU just to support the special cloud scenario.